mvg/plotting.py

```diff
@@ -201,15 +209,31 @@ def modes_over_time(
     # Create rectangular patch for timestamp
     datalist = data["labels"].tolist()
     uncertlist = data["uncertain"].tolist()
-    for i, val in enumerate(datalist):
+
+    ts_range = data["timestamps"].iloc[-1] - data["timestamps"].iloc[0]
+
+    # Collect the indices of data where modes change plus start and end points
+    interval_list = (
+        [0]
+        + [i for i in range(1, len(datalist)) if datalist[i] != datalist[i - 1]]
+        + [len(datalist) - 1]
+    )
+    for idx in range(len(interval_list) - 1):
         # gray border around uncertains
-        if uncertlist[i]:
-            ecol = -2
-        else:
-            ecol = val
+        i = interval_list[idx]
+        val = datalist[i]
+        ecol = -2 if uncertlist[i] else val
```

Comment:
This will not correctly display the uncertain points
![new_vs_old](https://user-images.githubusercontent.com/46282773/122712390-22f55c80-d264-11eb-9f64-2dfc60535a11.png)